lib/config.js

```diff
@@ -1,14 +1,43 @@
+'use strict';
+
 var config = {
-    brand: 'Discord',
-    token: process.env.OAUTH_TOKEN,
-    trackingID: process.env.TRACKING_ID,
-    commentWait: process.env.COMMENT_WAIT || 250,
-    commentAttempts: process.env.COMMENT_ATTEMPTS || 10,
-
-    // Used when no environment variable (process.env.PORT) is set
-    port: 3000,
-    host: 'localhost',
-    protocol: 'http:'
+    def: {
+        brand: 'Discord',
+        token: process.env.OAUTH_TOKEN,
+        commentAttempts: process.env.COMMENT_ATTEMPTS || 10,
+    },
+    development: {
+        port: 3000,
+        redisURL: 'redis://localhost',
+        databaseURL: 'postgres://localhost/discord_test'
+    },
+    test: {
+        protocol: 'http:',
+        host: 'localhost',
+        port: 3000,
+        trackingID: '654321',
+        commentWait: 0,
+        redisURL: 'redis://localhost',
+        databaseURL: 'postgres://localhost/discord_test',
+        database: 'discord_test'
+    },
+    production: {
+        port: process.env.PORT,
+        trackingID: process.env.TRACKING_ID,
+        commentWait: process.env.COMMENT_WAIT || 250,
+        redisURL: process.env.REDIS_URL,
+        databaseURL: process.env.DATABASE_URL
+    }
```

Comment:
Both approaches define configuration in source code; there's really no other way to do it.

The difference is in the original comment - i.e., putting named groups of specific deploys (`development`, `test`, `production`, etc.) into code does not scale cleanly as more deploys & environments ('stage', 'groovecoder-stage', 'contributor-xyz-stage', etc.) may need their own config values.

I **can** (and do) run a `groovecoder-stage` deployment of discord and override the config values via environment variables, but at that point I'm really just (ab)using the `production` config grouping, because it's doing the Right Thing™. i.e., it's injecting config from the environment into the app for run-time ([during the "release" phase of 12factor](http://12factor.net/build-release-run)).

So really, the named groups put more than just configuration into the source code. They also put a map/hierarchy/catalog of deployment environments into the source code, which then has to be maintained. :cry: